lib/mariaex/connection.ex

```diff
@@ -187,6 +187,9 @@ defmodule Mariaex.Connection do
     end
   end
 
+  def handle_info({:tcp_closed, _}, s) do
+    raise %Mariaex.Error{message: "Unable to connect to database"}
+  end
```

Comment:
If we should match Postgrex the function should return `{:stop, %Mariaex.Error{message: "connection closed"}, s}`.